File: 0198-house-robber/solution.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int prev = 0, curr = 0, next = 0;
        // prev and curr are DP memoization

        for (auto& i : nums) {
            next = max(prev+i, curr);
            prev = curr;
            curr = next;
        }

        return curr;
    }
};

File: 0198-house-robber/solution.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        a, b = 0, 0

        # a, b, i, i+1, ...
        for i in nums:
            temp = max(a+i, b)

            # update values for next iteration
            a = b
            b = temp
            # i is being incremented in the loop itself.
        
        return b
